### PR TITLE
SOLR-14829: Improve documentation for Request Handlers in RefGuide and solrconfig.xml

### PR DESCRIPTION
**Pull request by arafalov**
# Description

The information in the Guide was not flowing and confusing, the sections on defaults were as if they were for Search Handlers only, not for all and there were not enough forward references to actually understand the available options.

Additionally, the shipped solrconfig.xml information was completely out of date and even misleading (by pretending to be the source of truth) and additional information links were going to the wiki, which were either out of date or completely dead.

# Solution

Completely refactor the relevant Ref Guide page (git does not even show diff by default). 
Introduced a lot more cross-referencing and forward-referencing, including to URPs as parallel to Search Components.

Pointed solrconfig.xml links to RefGuide (latest-version redirects) and removed sections that were duplicating Ref Guide content (and falling out of sync).

# Tests

The changes were all documentation, but I double-checked that both configsets are still usable after changes made (no XML errors).

# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [X] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [X] I have developed this patch against the `master` branch.
- [X] I have run `./gradlew check`.
- [X] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by arafalov** [comment]
It looks like merge have gone haywire and duplicated records. But the changes all look correct and the "Squash and Merge" will flatten it all anyway.

File: solr/solr-ref-guide/src/requesthandlers-and-searchcomponents-in-solrconfig.adoc [hunk 1]
```diff
@@ -18,34 +18,84 @@
...
 
-== Request Handlers
+All of this multi-layered configuration, may be verified via  <<config-api.adoc#config-api,Config API>>.
 
-Every request handler is defined with a name and a class. The name of the request handler is referenced with the request to Solr, typically as a path. For example, if Solr is installed at `\http://localhost:8983/solr/` and you have a collection named "gettingstarted", you can make a request that looks like this:
+Defining your own config handlers is often a useful way to provide defaults and advanced configuration to support business cases and simplify client API.
+At the same time, using every single option explained in this guide, will most certainly cause some confusion about which parameter is actually used when.
+
+== Defining and calling Request Handlers
+
```
Comment (ctargett) [inline solr/solr-ref-guide/src/requesthandlers-and-searchcomponents-in-solrconfig.adoc:35]: Section headings should be in Headline Case (all main words capitalized)

File: solr/solr-ref-guide/src/requesthandlers-and-searchcomponents-in-solrconfig.adoc [hunk 2]
```diff
@@ -18,34 +18,84 @@
...
+  "query" : "memory",
+  "filter" : "inStock:true"
+}'
+----
 
-The primary request handler defined with Solr by default is the "SearchHandler", which handles search queries. The request handler is defined, and then a list of defaults for the handler are defined with a `defaults` list.
+Either way, the parameters are extracted and combined with other options explained below.
 
-For example, in the default `solrconfig.xml`, the first request handler defined looks like this:
+=== defaults, appends, and invariants
```
Comment (ctargett) [inline solr/solr-ref-guide/src/requesthandlers-and-searchcomponents-in-solrconfig.adoc:93]: I think these should be capitalized. In the sub-sections here you did not capitalize "defaults" but you did "Appends" - it should be consistent, and even though those terms are the XML names we can also refer to the concepts of default and append, etc, as words and capitalize them in headings.

File: solr/solr-ref-guide/src/common-query-parameters.adoc [hunk 3]
```diff
@@ -307,11 +307,13 @@ The `echoParams` parameter controls what information about request parameters is
 
 The `echoParams` parameter accepts the following values:
 
-* `explicit`: This is the default value. Only parameters included in the actual request, plus the `_` parameter (which is a 64-bit numeric timestamp) will be added to the `params` section of the response header.
+* `explicit`: Only parameters included in the actual request, plus the `_` parameter (which is a 64-bit numeric timestamp) will be added to the `params` section of the response header.
```
Comment (dsmiley) [inline solr/solr-ref-guide/src/common-query-parameters.adoc:310]: @gerlowskija this reminds me... (off topic of this PR), do you know where and why "_" parameter is added?  If it's for query identification purposes then it's duplicitous with the "rid" you added.  I'm having trouble searching for it.

**Review by dsmiley — APPROVED** [review]

File: solr/solr-ref-guide/src/common-query-parameters.adoc (hunk 3 above)
Comment (gerlowskija) [reply solr/solr-ref-guide/src/common-query-parameters.adoc:310]: No, I don't know anything about it.  This is the first I've heard of an underscore param.

It may be duplicitous with the 'rid' stuff I added recently, but that's not necessarily the case.  e.g. There's nothing here that suggests that a '_' value gets passed without change to shard-requests, afaict.  But it's worth following up on for sure, thanks for pointing it out @dsmiley 

> I'm having trouble searching for it.
No kidding!  Whoever put this in didn't have searchability in mind, that's for sure haha.

Comment (dsmiley) [reply solr/solr-ref-guide/src/common-query-parameters.adoc:310]: I did some digging and discovered that the admin UI will add this underscore.  See `services.js` which does this via `Date.now()` all over the place.  I don't know what purpose it has; git blame shows it was added with that whole UI refactor of it's day but perhaps it predated it?  AFAICT Solr isn't doing anything with it Solr side; I set multiple conditional debugger breakpoints across SolrParams subclasses but no where is "_" requested.  My suspicion is that it was added to foil HTTP cache attempts?

I did some JIRA digging and found: https://issues.apache.org/jira/browse/SOLR-4311 which has a patch with the underscores but it was for version, not the timestamp.  Given the underscore's use to defeat bad caching in that scenario, if I keep digging, I'll probably find it.

In short, I don't think we should document this underscore param in the ref guide.  It's not a special param to Solr; it's an oddity of our current admin UI.

Comment (dsmiley) [reply solr/solr-ref-guide/src/common-query-parameters.adoc:310]: @arafalov can you please remove the '_' reference here in the docs?  I think what it says about this is false; I suspect it was always false.

Comment (arafalov) [reply solr/solr-ref-guide/src/common-query-parameters.adoc:310]: Will do. I was busy on the other project, but will finish this one very soon.
